components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -1539,9 +1560,52 @@ func (p *pg) DeleteProject(ctx context.Context, id string) error {
 
 	err = p.singleRowResultOrNotFoundErr(res)
 	if err != nil {
+		// don't error if the project is already in the graveyard.
+		// this is necessary since the status might not have been reported
+		// to cereal on the first attempt, in which case we want this
+		// function to be idempotent.
+		if err == storage_errors.ErrNotFound {
```

Comment:
Thinking a bit about this, what happens if authz-service goes down while a project purge is in progress? Some objects may still have the deleted project; that's not good, but not a calamity, and would be difficult to triage. However, one thing I think we should do: delete any entries in `iam_projects_graveyard` on service startup, and add an entry in the log if there were any to delete.

Reply:
Not sure I follow exactly what you are concerned about here. Let's chat after standup tomorrow? Don't think removing things from the graveyard is gonna be the correct solution.